In [1]:
import os
import sys
import time
import json
import shutil
import warnings
warnings.filterwarnings('ignore')
import argparse
import numpy as np
import pandas as pd
from tqdm import tqdm
from sklearn.metrics import confusion_matrix
from sklearn.metrics import accuracy_score, f1_score, roc_auc_score, matthews_corrcoef
from imblearn.metrics import geometric_mean_score
from metrics import *
import matplotlib.pyplot as plt

import torch
import torch.nn as nn
import torch.optim as optim

from dataset import *
from utils import *
from models import *

In [2]:
cov = pd.read_csv("../SARS-SAbDab_Shaun/CoV-AbDab_extract.csv")
cov

Index      AB_name  Class                              Paratope  \
0        1         m396      0        SYTWGGITILIANYQRDTVMGGSKWDSSDY   
1        2          80R      0  SYAVISYDRDRSYDRVRSNDSTRSGSGSDFTTSRSW   
2        3       F26G19      0                YVSYTTYRYPGNDTTTEGIPQL   
3        4        1AF10      0            NWRENRYDPYYAMDIQHYQQTDSWPT   
4        5          6D3      1                WHEDDYINYEYGLLAQNNFYTY   
..     ...          ...    ...                                   ...   
305    306      scFv_E4      1    VSGVTSSNYYPGGSYRDLSEKDVIQGISYSGQNS   
306    307        N3113      1             YEALEWSYHSRTYNPSWASGSTGDW   
307    308      N3113.1      1                       QWYSRYNPSWASGST   
308    309  RBD-chAb-15      1                       GHWTTWGGSTNGYYV   
309    310       BD-813      1         VEFISANYYSGGSFRLPTYDSISYHDTTP   

                                       Epitope  
0                      FSTFRGDEVRNNGQPTYGVGYQY  
1              RDVGGYNLYLFKGFNCYFLQSYGQPTYGVYQ  
2                         AFSTGDRWNNNPTYGVGYQY  
3    FIVKNKCVFTGVLTDIATDLDITPCSFGGSYQGVNRAGCIG  
4                                  NVFAEHVNNSS  
..                                         ...  
305       RDQTGNDYYLFRKSNYQAGSTFNYFPLQSYGYGVGY  
306         VATRFASYAWNKSKGGNYNYLRITEINGVKFLQS  
307        ATRFASYANKDKVGNYNYLYITEINGVKFPLQSYG  
308                   YASFSTFKRGDRAWNTYGVGYQPY  
309             RDERQTGNDYYLFRKSNYQAGSTFNYQYGY  

[310 rows x 5 columns]

In [18]:
part = cov.sample(frac=0.1, random_state=42)
part.shape

(31, 5)

In [7]:
len(cov[cov["Class"]==1]["Paratope"].unique()), len(cov[cov["Class"]==1]["Epitope"].unique())

(228, 226)

In [10]:
len(cov[cov["Class"]==1]), len(cov[cov["Class"]==0])

(228, 82)

228 unique paratopes, 226 unique epitopes, 228 binding pairs, 82 non-binding pairs

In [15]:
228+225*9

2253